### PR TITLE
BLD: Allow setup to run without numpy in environment.

### PR DESCRIPTION
**Pull request by ehebert**
Defer the import of numpy until the time of extension compilation, so that
the setup's `install_requires` can partake in the dependency resolution chain.

This was noticed when building out a fresh enviroment which did not contain
numpy, resulting in a failure to import warp_prism's `setup.py`.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -22,6 +21,24 @@
     'Topic :: Scientific/Engineering',
 ]
 
+class BuildExtWithNumpyCommand(build_ext):
```
Comment (llllllllll) [inline setup.py]: This stuff is much more complicated than what we had before, what is wrong with the simple two step install?

@ssanderson has expressed a desire to remove this stuff from zipline even.

Comment (ehebert) [reply setup.py]: I had read your comment here https://github.com/quantopian/databazaar/pull/307#discussion_r144362567 as a request to move both numpy and Cython. I may have misread that, with you only meaning to move the `Cython` requirement.

Comment (ehebert) [reply setup.py]: However, a point against the current version of `setup.py` which does the upfront import of `numpy` is that it prevents any extra tooling like `pip-compile`  from working in a fresh environment.

Comment (llllllllll) [reply setup.py]: you just need to install numpy, which you need to do anyways.

**Comment by ehebert** [comment]
Closing, based on a discussion with @llllllllll where he advocated for not overcomplicating the `setup.py`.

**Pull request closed without merging**